proxy/http/ForwardedConfig.cc

```diff
@@ -0,0 +1,173 @@
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <bitset>
+#include <string>
+#include <cctype>
+
+#include <ts/string_view.h>
+#include <ts/MemView.h>
+
+#include <HttpConfig.h>
+
+namespace
+{
+class BadOptionsErrMsg
+{
+public:
+  // Construct with referece to string that will contain error message.
+  //
+  BadOptionsErrMsg(std::string &err) : _err(err), _count(0) {}
+
+  // Add a bad option.
+  //
+  void
+  add(ts::StringView badOpt)
+  {
+    if (_count == 0) {
+      _err   = _quote(badOpt);
+      _count = 1;
+    } else if (_count == 1) {
+      _saveLast = badOpt;
+      _count    = 2;
+    } else {
+      _err += ", " + _quote(_saveLast);
+      _saveLast = badOpt;
+      ++_count;
+    }
+  }
+
+  void
+  done()
+  {
+    if (_count == 0) {
+      _err.clear();
+    } else if (_count == 1) {
+      _err += " is a bad option.";
+    } else {
+      _err += " and " + _quote(_saveLast) + " are bad options.";
+    }
+  }
+
+private:
+  std::string
+  _quote(ts::StringView sv)
+  {
+    return std::string("\"") + std::string(sv.begin(), sv.size()) + "\"";
+  }
+
+  std::string &_err;
+
+  ts::StringView _saveLast;
+
+  int _count;
+};
+
+// Compare a StringView to a nul-termimated string, converting the StringView to lower case and ignoring whitespace in it.
+//
+bool
+eqIgnoreCaseWs(ts::StringView sv, const char *target)
+{
+  const char *s = sv.begin();
+
+  std::size_t skip = 0;
+  std::size_t i    = 0;
+
+  while ((i + skip) < sv.size()) {
+    if (std::isspace(s[i + skip])) {
+      ++skip;
+    } else if (std::tolower(s[i + skip]) != target[i]) {
+      return false;
+    } else {
+      ++i;
+    }
+  }
+
+  return target[i] == '\0';
+}
+
+} // end anonymous namespace
+
+namespace HttpForwarded
+{
+OptionBitSet
+optStrToBitset(ts::string_view optConfigStr, std::string &error)
+{
+  const ts::StringView Delimiters(":|");
+
+  OptionBitSet optBS;
+
+  // Convert to TS StringView to be able to use parsing members.
+  //
+  ts::StringView oCS(optConfigStr.data(), optConfigStr.size());
+
+  error.clear();
+
+  if (eqIgnoreCaseWs(oCS, "none")) {
+    return OptionBitSet();
+  }
+
+  BadOptionsErrMsg em(error);
+
+  do {
+    ts::StringView optStr = oCS.extractPrefix(Delimiters);
+
+#undef X
```

Comment:
This seems like a good place for a lambda, rather than a #define.

Reply:
I don't see how to use a lambda here.   I could cut-and-paste to get rid of the macro, but I need more details to know how you want a lambda used here.